File: javascript/arrays/iterating_over_arrays_and_updating_elements.js
#!/usr/bin/env node

array = [1, 3, 5];

console.log("Before:");
console.log(array);

for (let i = 0; i < array.length; i++) {
  array[i] *= 2;
}

console.log("After:");
console.log(array);

console.log();

array = [1, 3, 5];

console.log("Before:");
console.log(array);

for (let key in array) {
  array[key] *= 3;
}

console.log("After:");
console.log(array);

console.log();

array = [1, 3, 5];

console.log("Before:");
console.log(array);

array.forEach(function(value, key) {
  array[key] *= 4;
});

console.log("After:");
console.log(array);

console.log();

File: perl/arrays/arrays/iterating_over_arrays_and_updating_elements.pl
#!/usr/bin/perl

use strict;
use warnings;

my @array = (1, 3, 5);

print("Before: @array\n");

for (my $i = 0; $i < @array; $i++) {
  $array[$i] *= 2;
}

print("After: @array\n");

print("\n");

# for can be used instead of foreach also

@array = (1, 3, 5);

print("Before: @array\n");

foreach (0..(@array - 1)) {
  $array[$_] *= 2;
}

print("After: @array\n");

print("\n");

@array = (1, 3, 5);

print("Before: @array\n");

foreach my $key (0..(@array - 1)) {
  $array[$key] *= 2;
}

print("After: @array\n");

print("\n");

@array = (1, 3, 5);

print("Before: @array\n");

foreach (@array) {
  $_ *= 2;
}

print("After: @array\n");

print("\n");

@array = (1, 3, 5);

print("Before: @array\n");

foreach my $element (@array) {
  $element *= 2;
}

print("After: @array\n");

print("\n");
